### PR TITLE
Patient search system - autocomplete

### PR DESCRIPTION
**Pull request by Johngtka**
Added material autocomplete to search for registered patients

brave_sxFKzwh7pz

capturing some server error:
brave_op253oiCXM

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,3 +1 @@
-<div>
-  this is a app component!! and this is a title: {{ 'APP.TITLE' | translate }}
-</div>
\ No newline at end of file
+<app-usersearch></app-usersearch>
```
Comment (macrog) [inline src/app/app.component.html:1]: prettier this file please

File: src/app/app.component.ts [hunk 2]
```diff
@@ -2,7 +2,6 @@ import { Component } from '@angular/core'
 
 @Component({
   selector: 'app-root',
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css'],
+  templateUrl: './app.component.html'
```
Comment (macrog) [inline src/app/app.component.ts:5]: why did you remove styles link of the component?

File: src/app/usersearch/usersearch.component.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-usersearch',
+  templateUrl: './usersearch.component.html'
```
Comment (macrog) [inline src/app/usersearch/usersearch.component.ts:5]: where are styles?

File: src/app/usersearch/usersearch.component.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+})
+export class UsersearchComponent {
```
Comment (macrog) [inline src/app/usersearch/usersearch.component.ts:7]: why UsersearchComponent ? it should be called PatientSeachComponent?

File: src/app/usersearch/usersearch.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<p>usersearch works!</p>
```
Comment (macrog) [inline src/app/usersearch/usersearch.component.html:1]: where is autocomplete?

**Review by macrog — CHANGES_REQUESTED** [review]
?¿

File: src/app/app.component.ts (hunk 2 above)
Comment (Johngtka) [reply src/app/app.component.ts:5]: because we dont need components styles, we will use the global styles in app

File: src/app/usersearch/usersearch.component.html (hunk 5 above)
Comment (Johngtka) [reply src/app/usersearch/usersearch.component.html:1]: it will be

File: src/app/usersearch/usersearch.component.ts (hunk 4 above)
Comment (Johngtka) [reply src/app/usersearch/usersearch.component.ts:7]: done

File: src/app/app.component.ts (hunk 2 above)
Comment (macrog) [reply src/app/app.component.ts:5]: no, add them, each file will have it's own styles, even if they are empty

File: src/app/app.component.html [hunk 6]
```diff
@@ -1,3 +1 @@
-<div>
-  this is a app component!! and this is a title: {{ 'APP.TITLE' | translate }}
-</div>
\ No newline at end of file
+<app-patient-search></app-patient-search>
```
Comment (macrog) [inline src/app/app.component.html:1]: missing empty line at the end of file, please format file with prettier

File: src/app/app.component.spec.ts [hunk 7]
```diff
@@ -14,12 +14,6 @@ describe('AppComponent', () => {
...
   })
 
-  it(`should have as title 'Biomagnethic-App'`, () => {
-    const fixture = TestBed.createComponent(AppComponent)
-    const app = fixture.componentInstance
-    expect(app.title).toEqual('Biomagnethic-App')
-  })
-
   it('should render title', () => {
     const fixture = TestBed.createComponent(AppComponent)
```
Comment (macrog) [inline src/app/app.component.spec.ts:18]: please stop doing changes to `*.spec.ts` files, we will fix them all at the end

File: src/app/app.component.ts [hunk 8]
```diff
@@ -3,6 +3,6 @@ import { Component } from '@angular/core'
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css'],
+  styleUrls: ['./app.styles.css']
```
Comment (macrog) [inline src/app/app.component.ts:6]: the file name should be `app.component.css` not `app.style.css`

File: src/app/app.module.ts [hunk 9]
```diff
@@ -3,22 +3,39 @@ import { BrowserModule } from '@angular/platform-browser'
...
+
+import { MatAutocompleteModule } from '@angular/material/autocomplete';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input'
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+
 import { AppComponent } from './app.component';
+import { PatientSearchComponent } from './patient-search/patient-search.component';
+import { PatientService } from './services/patient-service';
```
Comment (macrog) [inline src/app/app.module.ts:20]: please organise imports in 3 blocks, 1st angular 2nd external libraries and 3rd our application imports

File: src/app/patient-search/patient-search.component.ts [hunk 10]
```diff
@@ -0,0 +1,70 @@
+import { Component } from '@angular/core';
+import { OnInit } from '@angular/core';
+
+import { FormControl } from '@angular/forms';
+import { debounceTime } from 'rxjs/operators';
+import { tap } from 'rxjs/operators';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { filter } from 'rxjs/operators';
+
+import { PatientService } from '../services/patient-service';
```
Comment (macrog) [inline src/app/patient-search/patient-search.component.ts:10]: this is a very well organise imports, good job here 👍

File: src/app/patient-search/patient-search.component.ts [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+export interface User {
+  name: string;
+}
```
Comment (macrog) [inline src/app/patient-search/patient-search.component.ts:14]: User interface is not used, it can be removed

File: src/app/services/patient-service.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { Observable } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
```
Comment (macrog) [inline src/app/services/patient-service.ts:3]: this imports could be organised better

File: src/app/services/patient-service.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class PatientService {
+  constructor(private http: HttpClient) { }
+  PatientSearch(query: string): Observable<any> {
```
Comment (macrog) [inline src/app/services/patient-service.ts:9]: `PatientSearch` should be `patientSearch` -> always the same, camel case for variables and methods, starts on lowercase

File: src/index.html [hunk 14]
```diff
@@ -1,16 +1,15 @@
...
 
-  <body class="mat-typography">
-    <app-root></app-root>
-  </body>
-</html>
+<body>
+  <app-root></app-root>
+</body>
+
+</html>
```
Comment (macrog) [inline src/index.html:15]: prettier this file, missing empty line at the end of file!

File: src/app/patient-search/patient.styles.css [hunk 15]
```diff
@@ -0,0 +1,11 @@
+.wrapper {
+  width: 40%;
+  background-color: rgb(164, 164, 163);
+  margin-left: auto;
+  margin-right: auto;
```
Comment (macrog) [inline src/app/patient-search/patient.styles.css:5]: can be removed, not needed background-color, width, margin-left and margin-right

File: src/app/patient-search/patient.styles.css [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+  text-align: center;
+}
+
+.patient-full-width {
+  text-align: center;
+}
```
Comment (macrog) [inline src/app/patient-search/patient.styles.css:11]: prettier this file, missing empty line at the end

File: src/app/patient-search/patient-search.component.html [hunk 17]
```diff
@@ -0,0 +1,28 @@
+<div class="wrapper">
+  <h2>Patient Search</h2>
```
Comment (macrog) [inline src/app/patient-search/patient-search.component.html:2]: remove <h2>

**Review by macrog — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts (hunk 9 above)
Comment (Johngtka) [reply src/app/app.module.ts:20]: done

File: src/app/app.component.spec.ts (hunk 7 above)
Comment (Johngtka) [reply src/app/app.component.spec.ts:18]: restored

File: src/app/services/patient-service.ts (hunk 13 above)
Comment (Johngtka) [reply src/app/services/patient-service.ts:9]: done

File: src/index.html (hunk 14 above)
Comment (Johngtka) [reply src/index.html]: done

**Comment by macrog** [comment]
what are these:
image

you cannot make commits with the same message or with just a "." 😆 please make it better next time

**Review by macrog — APPROVED** [review]
Great work, I know it wasn't easy!! 🍾 🍺 🎖️

**Comment by macrog** [comment]
@Johngtka squash and merge message? I would like you to try provide yours when you open a PR.

`feat: Allow searching for a specific patient (#7)`
